src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt

```diff
@@ -131,9 +143,38 @@ class PrisonerSearchService(
       throw BadRequestException("Invalid search  - please provide a minimum of 1 and a maximum of 200 prisoner numbers")
     }
 
-    queryBy(prisonerListCriteria) { matchByIds(it) } onMatch { return it.matches }
+    queryBy(prisonerListCriteria) { matchByIds(it) } onMatch {
+      customEventForFindByListOfPrisonerNumbers(prisonerListCriteria.prisonerNumbers.size, it.matches.size)
+      return it.matches }
     return emptyList()
   }
+
+  private fun customEventForFindBySearchCriteria(searchCriteria: SearchCriteria, numberOfResults: Int) {
+    val propertiesMap = mapOf(
+      "user" to authenticationFacade.currentUsername(),
+      "clientId" to authenticationFacade.currentClientId(),
+      "lastname" to searchCriteria.lastName,
+      "firstname" to searchCriteria.firstName,
+      "prisonId" to searchCriteria.prisonId,
+      "prisoner identifier" to searchCriteria.prisonerIdentifier,
+      "alias" to searchCriteria.includeAliases.toString()
+    )
+    val metricsMap = mapOf(
+    "numberOfResults" to numberOfResults.toDouble()
+    )
+    telemetryClient.trackEvent("FindByCriteria", propertiesMap, metricsMap)
+  }
+
+  private fun customEventForFindByListOfPrisonerNumbers(prisonerListNumber: Int, numberOfResults: Int) {
+    val logMap = mapOf(
+      "user" to authenticationFacade.currentUsername(),
+      "clientId" to authenticationFacade.currentClientId(),
+      "numberOfPrisonerIds" to prisonerListNumber.toString(),
+      "numberOfResults" to numberOfResults.toString()
```

Comment:
would be better for the `numberOfResults` to be in the metrics, like in the other log event